### PR TITLE
feat(op): signature encoding

### PR DESCRIPTION
**Pull request by emhane**
Adds back support for encoding null signature, that was reverted in https://github.com/paradigmxyz/reth/pull/7724

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
do we need all of those or now just the v value special case? for != 27/28?

**Comment by emhane** [comment]
> do we need all of those or now just the v value special case? for != 27/28?

yes we need all, I'm running import on branch https://github.com/paradigmxyz/reth/tree/emhane/mainnet-op, and have encountered both of those errors.

File: crates/primitives/src/transaction/signature.rs [hunk 1]
```diff
@@ -98,6 +102,10 @@ impl Signature {
         if v < 35 {
             // non-EIP-155 legacy scheme, v = 27 for even y-parity, v = 28 for odd y-parity
             if v != 27 && v != 28 {
+                #[cfg(feature = "optimism")]
+                if v == 0 && r == U256::ZERO && s == U256::ZERO {
```
Comment (rkrasiuk) [inline crates/primitives/src/transaction/signature.rs:106]: nit
```suggestion
                if v == 0 && r.is_zero() && s.is_zero() {
```

**Review by rkrasiuk — APPROVED** [review]

**Review by mattsse — CHANGES_REQUESTED** [review]
this needs docs, why we're doing this

**Review by mattsse — APPROVED** [review]

**Pull request merged**